### PR TITLE
Update examples to AWS v6

### PR DESCRIPTION
**Pull request by mnlumi**
Resolves https://github.com/pulumi/pulumi-aws/issues/2621

### CODE REVIEW

**Comment by mnlumi** [comment]
Setting to alpha release for testing.

**Comment by mnlumi** [comment]
FYI @cnunciato I've got this set to the alpha release for testing right now. They will be updated to v6 once the actual release is cut.

File: testing-unit-go/main_test.go [hunk 1]
```diff
@@ -53,9 +53,9 @@ func TestInfrastructure(t *testing.T) {
 		// Test if the instance is configured with user_data.
 		pulumi.All(infra.server.URN(), infra.server.UserData).ApplyT(func(all []interface{}) error {
 			urn := all[0].(pulumi.URN)
-			userData := all[1].(*string)
+			userData := all[1].(string)
```
Comment (t0yv0) [inline testing-unit-go/main_test.go:56]: Breaking change somewhere?

**Comment by t0yv0** [comment]
A new [v6.0.0-alpha.1](https://github.com/pulumi/pulumi-aws/releases/tag/untagged-72a25b9a4bca1aa05af3) alpha is out, let me see if I can update quickly.

**Comment by mnlumi** [comment]
Equinix failures are unrelated to AWS 6.0. The test on `TestAccAwsPyEc2Provisioner` is also unrelated and is being handled separately via https://github.com/pulumi/examples/issues/1491 and https://github.com/pulumi/examples/pull/1492.

**Review by iwahbe — APPROVED** [review]

**Review by thomas11 — CHANGES_REQUESTED** [review]
I checked out the PR branch and ran some searches and the following seem to be missing:

```
> rg 'aws.+<6.0'
aws-py-hub-and-spoke-network/requirements.txt
2:pulumi_aws>=5.0.0,<6.0.0

> rg 'pulumi-aws/sdk/v5'
aws-go-fargate/go.mod
6:	github.com/pulumi/pulumi-aws/sdk/v5 v5.41.0
47:	github.com/pulumi/pulumi-aws/sdk/v5 v5.16.2 // indirect

aws-go-fargate/main.go
8:	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ec2"
9:	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ecr"
10:	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ecs"
11:	elb "github.com/pulumi/pulumi-aws/sdk/v5/go/aws/elasticloadbalancingv2"
12:	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/iam"

aws-go-fargate/go.sum
[...]
```

**Comment by mnlumi** [comment]
Ive intentionally excluded updates to examples that also use awsx. I will update those after the next awsx release that takes in aws 6.0.

@thomas11 The programs you mentioned both use awsx.
- aws-py-hub-and-spoke-network
- aws-go-fargate

**Review by thomas11 — APPROVED** [review]

**Pull request merged**